modules/internal/BytesCasts.chpl

```diff
@@ -246,9 +246,21 @@ module BytesCasts {
   //
   // complex
   //
+  @edition(first="preview")
+  operator :(x: chpl_anycomplex, type t:bytes) {
+    return chpl_bytesCastHelper(x);
+  }
+
+  @edition(last="2.0")
   operator :(x: chpl_anycomplex, type t:bytes) {
     if isNan(x.re) || isNan(x.im) then
       return b"nan";
+    return chpl_bytesCastHelper(x);
+  }
+
+  // TODO: when 2.0 edition is removed, merge this back into the `:` operator
+  // above
```

Comment:
Refactoring BytesCasts and StringCasts into a single helper module is a long-standing wish for me. While making this change, one can also consider adding this as a generic helper into `modules/internal/BytesStringCommon.chpl` as a way to avoid adding more duplicate code.

The drawback is that we'll have inconsistent implementations between different casts.